inventory.lua

```diff
@@ -83,63 +102,109 @@ minetest.register_node("digilines:chest", {
 			return not pipeworks.connects.facingFront(i,param2)
 		end,
 		input_inventory = "main",
-		can_insert = function(pos, _, stack)
-			return can_insert(pos, stack)
+		can_insert = function(pos, _, stack, vel)
+			local ret = minetest.get_meta(pos):get_inventory():room_for_item("main", stack)
+			if not ret then
+				-- The stack cannot be accepted. It will never be passed to
+				-- insert_object, but it should be reported as a toverflow.
+				-- Here, vel = direction item is moving, which is into side.
+				local side = vector.multiply(vel, -1)
+				send_message(pos, "toverflow", stack, side)
+			end
+			return ret
 		end,
-		insert_object = function(pos, _, stack)
-			local inv = minetest.get_meta(pos):get_inventory()
-			local leftover = inv:add_item("main", stack)
-			local count = leftover:get_count()
-			if count == 0 then
-				local derpstack = stack:get_name()..' 1'
-				if not inv:room_for_item("main", derpstack) then
-					-- when you can't put a single more of whatever you just put,
-					-- you'll get a put for it, then a full
-					sendMessage(pos,"full "..maybeString(stack)..' '..tostring(count))
-				end
-			else
-				-- this happens when the chest has received two stacks in a row and
-				-- filled up exactly with the first one.
-				-- You get a put for the first stack, a put for the second
-				-- and then a overflow with the first in stack and the second in leftover
-				-- and NO full?
-				sendMessage(pos,"overflow "..maybeString(stack)..' '..tostring(count))
+		insert_object = function(pos, _, stack, vel)
+			-- Here, vel = direction item is moving, which is into side.
```

Comment:
Then name it correspondingly.

Reply:
In this case `vel` is the name used by Pipeworks when calling the function, and it’s a rather weird hybrid between a direction and a velocity. It’s a table containing `x`, `y`, and `z` components which are normalized to have a magnitude of 1, plus also a `speed` component containing the magnitude. Do you call that a direction or a velocity?

Reply:
`insert_object = function(pos, node, stack, direction)` is used everywhere in `pipeworks`. Well, except of the place it is called at...

Reply:
Oh, I didn’t notice. I only looked at the place it was called from. Fixed in 478c6c8964cd88241bf33218296de4af1f70bacf.